### PR TITLE
fix: use poetry provided requirements for docs

### LINKED ISSUE #1481: RTD builds failing
# Bug report

## What's wrong

RTD builds fail: possibly due to package management

## How it should be

RTD builds successfully

## System information

NA

### PR DESCRIPTION
**Pull request by ryangrose**
# I have made things!

Fixes the RTD builds by using a pip install instead of poetry. To do this, we export a requirements.txt from poetry to ensure the build uses its dependency management. Then RTD installs from that requirements file.

Note that this uses the [dev dependency group](https://python-poetry.org/docs/managing-dependencies#dependency-groups). We could probably minimize this by creating a doc specific dependency group in the future.

## Checklist

- [ x ] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [ ] I have created at least one test case for the changes I have made
- [ ] I have updated the documentation for the changes I have made
- [ x ] I have added my changes to the `CHANGELOG.md`

## Related issues

- Closes #1481 

🙏 Please, if you or your company finds `dry-python` valuable, help us sustain the project by sponsoring it transparently on https://github.com/sponsors/dry-python. As a thank you, your profile/company logo will be added to our main README which receives hundreds of unique visitors per day.

### CODE REVIEW

**Comment by ryangrose** [comment]
This is a WIP for now: not sure how else to test RTD builds

File: .readthedocs.yml [hunk 1]
```diff
@@ -11,9 +11,14 @@ build:
       - asdf install poetry latest
       - asdf global poetry latest
       - poetry config virtualenvs.create false
+      - poetry export --with dev --format=requirements.txt --output=requirements.txt
     post_install:
```
Comment (sobolevn) [inline .readthedocs.yml:15]: This can be deleted now

File: CHANGELOG.md [hunk 2]
```diff
@@ -14,6 +14,7 @@ See [0Ver](https://0ver.org/).
 - Fixes custom pickle protocol to handle `None` values gracefully
 - Removes broken drylabs.io link in README
 - Revises pointfree documentation
+- Fixes RTD builds
```
Comment (sobolevn) [inline CHANGELOG.md:17]: I don't think it is worth adding

Comment (ryangrose) [reply CHANGELOG.md]: Removed

**Review by sobolevn — APPROVED** [review]
Thanks a lot for your help!

**Pull request merged**